### PR TITLE
Create a new workflow around ELB that is separate from ASG

### PR DESCRIPTION
**Pull request by atayarani**
@Tapjoy/eng-group-ops

### CODE REVIEW

**Comment by atayarani** [comment]
@Tapjoy/eng-group-ops 

This is ready for review (note: this is being merged into an integration branch)

File: bin/tass [hunk 1]
```diff
@@ -37,10 +36,31 @@ when 'create'
...
+  unless confirm_config(aws_env, config, opts)
+    abort('Cannot continue if configuration is not correct.  Please fix.')
+  end
   Tapjoy::AutoscalingBootstrap::AutoscalingGroup.new.create(opts, config, aws_env, user_data)
+
+  if config.key?(:elb)
+    config[:elb].each do |elb|
+      elb.each do |elb_name, elb_config|
+        Tapjoy::AutoscalingBootstrap.elb_name = elb_name
+        Tapjoy::AutoscalingBootstrap.create_elb = true
```
Comment (StabbyCutyou) [inline bin/tass]: What if you just want to attach to existing elbs, not create + attach?

How does this design account for that?

File: lib/tapjoy/autoscaling_bootstrap/elb.rb [hunk 2]
```diff
@@ -2,16 +2,30 @@ module Tapjoy
...
-      # Initialize the class
-      def initialize
+      def initialize(elb_hash, clobber_elb, zones, security_groups)
+        elb_config = build_config(elb_hash, zones, security_groups)
+        check_valid_config(elb_config)
+
+        if exists && clobber_elb
+          delete
+        elsif exists
+          return
```
Comment (StabbyCutyou) [inline lib/tapjoy/autoscaling_bootstrap/elb.rb]: Don't quote me on this, but I feel like ```return``` is unnatural in initializers - The better approach might be to optionally choose to run create(elb_config), and let the method return naturally as it hits the last line.

**Comment by StabbyCutyou** [comment]
How does this fit with existing .tass scripts? Do those continue to work, and this is just a new API?

File: lib/tapjoy/autoscaling_bootstrap/elb.rb (hunk 2 above)
Comment (atayarani) [inline lib/tapjoy/autoscaling_bootstrap/elb.rb]: oops.  This was a placeholder.  I'll have a new update that should fix this and answer your previous question.

**Comment by atayarani** [comment]
This will break existing .tass scripts if they use ELBs.

**Comment by atayarani** [comment]
@StabbyCutyou PR has been updated

**Comment by StabbyCutyou** [comment]
Whats the plan about existing .tass scripts?

**Comment by atayarani** [comment]
Users will need to update them to match the new style.  It's unfortunately a breaking change, but the new version number should mitigate that.

**Comment by StabbyCutyou** [comment]
Only concern, so long as this works, is that it will break old .tass scripts. It might be worthwhile holding off on this until either those are ready for updating at the same time, or you provide a fallback mechanism to do the old behavior in tass for the time being?

**Comment by marcuswalser** [comment]
I think we should just update the existing .tass stuff immediately prior to
merging this guy. I don't think the uptake of tass is sufficient that we
need to provide a fallback mechanism.

**Comment by StabbyCutyou** [comment]
I'm fine with that option

**Comment by atayarani** [comment]
Since this is a merge to the integration branch, and not master, I'll move ahead with this merge.  Once I have the rest of the features included, there will be a bigger integration branch -> master merge including the version bump to 1.0.0

**Comment by marcuswalser** [comment]
Sounds good.

**Pull request merged**